lib/client/base.js

```diff
@@ -745,9 +745,16 @@
   HappnClient.prototype.reconnect = function (options) {
     var _this = this;
 
+    if (!this.reconnectAttempt) this.reconnectAttempt = 0;
+    var reconnectAttempt = ++this.reconnectAttempt;
+
     _this.authenticate(function (e) {
+      if (_this.reconnectAttempt != reconnectAttempt) return; // this is not the last attempt, ignore it
```

Comment:
however we could also consider simply using a late response received after retrying